sklearn/utils/tests/test_stats.py

```diff
@@ -96,3 +96,71 @@ def test_weighted_percentile_2d():
         _weighted_percentile(x_2d[:, i], w_2d[:, i]) for i in range(x_2d.shape[1])
     ]
     assert_allclose(w_median, p_axis_0)
+
+
+def test_weighted_percentile_nan_filtered():
+    """Test that calling _weighted_percentile on an array with nan values returns
+    the same results as calling _weighted_percentile on a filtered version of the data.
+    We test both with sample_weight of the same shape as the data and for
+    one-dimensional sample_weight."""
+
+    rng = np.random.RandomState(42)
+    array = rng.rand(10, 100)
+
+    array_with_nans = array.copy()
+    array_with_nans[rng.rand(*array_with_nans.shape) < 0.5] = np.nan
+    nan_mask = np.isnan(array_with_nans)
+
+    sample_weights = [rng.randint(1, 6, size=(10, 100)), rng.randint(1, 6, size=(10,))]
+
+    for sample_weight in sample_weights:
+        # Find the weighted percentile on the array with nans:
+        results = _weighted_percentile(array_with_nans, sample_weight, 30)
+
+        # Find the weighted percentile on the filtered array:
+        filtered_array = [
+            array[~nan_mask[:, col], col] for col in range(array.shape[1])
```

Comment:
We never need to access the original value before we replaced them by nans and the test of the code should not make the reader think otherwise. So better use:

```suggestion
            array_with_nans[~nan_mask[:, col], col] for col in range(array.shape[1])
```

which should give the same result.

Reply:
Now I understand it. Thank you.